docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,334 @@
+# The resulting weak form is given given as follows: Find $u \in \mathbb{U}$ such that
+# ```math
+#  \int_\Omega \nabla u \cdot \nabla \delta u \,\mathrm{d}\Omega - \int_{\Gamma^0} \llbracket u\rrbracket  \cdot \boldsymbol{\{\nabla \delta u\}} + \llbracket \delta u\rrbracket  \cdot \boldsymbol{\{\nabla u\}}  \,\mathrm{d}\Gamma^0 + \int_{\Gamma^0} \frac{\eta}{h} \llbracket u\rrbracket \cdot \llbracket \delta u\rrbracket   \,\mathrm{d}\Gamma^0 = \int_\Omega \delta u \,\mathrm{d}\Omega + \int_{\partial \Omega_N} (\nabla u \cdot \boldsymbol{n}) \delta u \,\mathrm{d} \partial \Omega_N,\\
+# ```
+# where $h$ is the characteristic mesh size (the maximum diameter of the cells), and $\eta$ is a large enough positive number independent of $h$ [Mu:2014:IP](@cite),
+# $\delta u \in \mathbb{T}$ is a test function, and where $\mathbb{U}$ and $\mathbb{T}$ are suitable
+# trial and test function sets, respectively.
+#
+# More details on DG formulations for elliptic problems can be found in [Cockburn:2002:unifiedanalysis](@cite)
+#-
+# ## Commented Program
+#
+# Now we solve the problem in Ferrite. What follows is a program spliced with comments.
+#md # The full program, without comments, can be found in the next [section](@ref heat_equation-DG-plain-program).
+#
+# First we load Ferrite and other packages, and generate grid just like the [heat equation tutorial](heat_equation.md)
+using Ferrite, SparseArrays
+dim = 2;
+grid = generate_grid(Quadrilateral, ntuple(_ -> 20, dim));
+# We construct the topology information which is used later for generating the sparsity pattern for stiffness matrix.
+topology = ExclusiveTopology(grid)
+
+# ### Trial and test functions
+# `CellValues`, `FaceValues`, and `InterfaceValues` facilitate the process of evaluating values and gradients of
+# test and trial functions (among other things). To define
+# these we need to specify an interpolation space for the shape functions.
+# We use `DiscontinuousLagrange` functions
+# based on the two-dimensional reference quadrilateral. We also define a quadrature rule based on
+# the same reference element. We combine the interpolation and the quadrature rule
+# to `CellValues` and `InterfaceValues` object. Note that `InterfaceValues` object contains two `FaceValues` objects which can be used individually.
+order = 1;
+ip = DiscontinuousLagrange{RefQuadrilateral, order}();
+qr = QuadratureRule{RefQuadrilateral}(2);
+# For `FaceValues` and `InterfaceValues` we use `FaceQuadratureRule`
+face_qr = FaceQuadratureRule{RefQuadrilateral}(2);
+cellvalues = CellValues(qr, ip);
+facevalues = FaceValues(face_qr, ip);
+interfacevalues = InterfaceValues(facevalues);
+# ### Penalty term parameters
+# We calculate the parameter $h$ as the maximum cell diameter.
```

Comment:
I think this does not matter in practice, but technically it should be the face diameter, right? In the referenced paper they seem to use $h_e$ for the face $e$ and $h_K$ for the element size, where only $h_e$ appears in their bilinear form (above eq. 2.3).

```suggestion
# We calculate the characteristic size $h$ as the maximum face diameter.
```

I also want to note that, I am not sure if my statement regarding *maximum diameter* is correct or if it should be the minimum diameter (from a theoretical point of view). Does not matter for the case in the example at least.

Furthermore, this computation also just works for structured grids. We should either add a note on this, or we just compute $h$ on the fly during the interface assembly for each face.